### PR TITLE
refactor: handle rejected tx & messages

### PR DESCRIPTION
**Pull request by namgold**

### CODE REVIEW

File: src/pages/AddLiquidityV2/index.tsx [hunk 1]
```diff
@@ -498,12 +502,18 @@ export default function AddLiquidity() {
...
-            console.error('Failed to send transaction', error)
             setAttemptingTxn(false)
-            // we only care if the error is something _other_ than the user rejected the tx
-            if (error?.code !== 4001) {
-              console.error(error)
-            }
+            // sending tx error, not tx execute error
+            const message = friendlyError(error)
+            console.error('Add liquidity error:', { message, error })
+            notify(
```
Comment (nguyenhoaidanh) [inline src/pages/AddLiquidityV2/index.tsx:509]: will show when user reject?

Comment (namgold) [reply src/pages/AddLiquidityV2/index.tsx:509]: Yes. Currently we not showing anything, quite bad in UX

File: src/pages/SwapV2/index.tsx [hunk 2]
```diff
@@ -311,7 +311,7 @@ export default function Swap() {
         setSwapState({ attemptingTxn: false, tradeToConfirm, showConfirm, swapErrorMessage: undefined, txHash: hash })
       })
       .catch(error => {
-        if (error?.code !== 4001 && error?.code !== 'ACTION_REJECTED') captureSwapError(error)
+        captureSwapError(error)
```
Comment (nguyenhoaidanh) [inline src/pages/SwapV2/index.tsx:314]: will capture when user reject?

Comment (namgold) [reply src/pages/SwapV2/index.tsx:314]: handled inside `captureSwapError` already

File: src/constants/connectors/utils.ts [hunk 3]
```diff
@@ -51,21 +48,24 @@ export enum ErrorCode {
...
-    error?.code === ErrorCode.ALPHA_WALLET_USER_REJECTED_REQUEST ||
+    error?.code === ErrorCode.ALPHA_WALLET_REJECTED_CODE ||
     error?.message === ErrorCode.ALPHA_WALLET_REJECTED ||
-    (connector === walletConnectV2 && error?.toString?.() === ErrorCode.WC_MODAL_CLOSED) ||
-    (connector === walletConnectV2 && error?.message === ErrorCode.CANCELED) ||
-    (connector === coinbaseWallet && error?.toString?.() === ErrorCode.CB_REJECTED_REQUEST)
+    error?.message === ErrorCode.WALLETCONNECT_MODAL_CLOSED ||
+    error?.message === ErrorCode.WALLETCONNECT_CANCELED ||
+    error?.message === ErrorCode.WALLETCONNECT_MODAL_CLOSED ||
+    rejectedPhrases.some(phrase => error?.message?.includes(phrase))
```
Comment (nguyenhoaidanh) [inline src/constants/connectors/utils.ts:69]: error?.message?.toLowerCase().includes

**Review by nguyenhoaidanh — APPROVED** [review]

File: src/components/SwapForm/AddMEVProtectionModal.tsx [hunk 4]
```diff
@@ -113,10 +112,9 @@ export default function AddMEVProtectionModal({ isOpen, onClose }: { isOpen: boo
         onClose?.()
         mixpanelHandler(MIXPANEL_TYPE.MEV_ADD_RESULT, { type: addingOption.name, result: 'success' })
       },
-      (connector: Connector, error: Error) => {
-        let reason = error?.message || 'Unknown reason'
-        if (didUserReject(connector, error)) reason = 'User rejected'
-        mixpanelHandler(MIXPANEL_TYPE.MEV_ADD_RESULT, { type: addingOption.name, result: 'fail', reason })
+      (error: Error) => {
+        const message = friendlyError(error)
+        mixpanelHandler(MIXPANEL_TYPE.MEV_ADD_RESULT, { type: addingOption.name, result: 'fail', reason: message })
```
Comment (viet-nv) [inline src/components/SwapForm/AddMEVProtectionModal.tsx:117]: should keep raw message on mixpanel. only need friendly message for end user

Comment (namgold) [reply src/components/SwapForm/AddMEVProtectionModal.tsx:117]: This is request from data analyst

**Pull request merged**